.github/workflows/release.yaml

```diff
@@ -0,0 +1,56 @@
+# Makes an automated release
+#
+# To trigger this workflow, ensure the CHANGELOG is up to date, then push the commit 
+# to be released to a `release/*` branch.
+#
+# e.g.
+#     git checkout -b release/v1.0.0
+#     git push
+#
+name: Release
+
+on:
+  push:
+    branches: [release/*]
+
+jobs:
+  create-release:
+    name: Create GitHub Release and tag
+    runs-on: ubuntu-latest
+    environment: Release
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cucumber-actions/create-release@v1.1.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  
+  publish-rubygem:
+    name: Publish Ruby Gem
+    needs: create-release
+    runs-on: ubuntu-latest
+    environment: Release
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.0.2
+          bundler-cache: true
+      - uses: cucumber-actions/publish-rubygem@v1.0.0
+        with:
+          rubygems_api_key: ${{ secrets.RUBYGEMS_API_KEY }}
+          working_directory: 'ruby'
+
+  publish-npm:
+    name: Publish NPM package
+    needs: create-release
+    runs-on: ubuntu-latest
+    environment: Release
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+      - uses: cucumber-actions/publish-npm@v1.0.0
```

Comment:
Can we use https://github.com/marketplace/actions/npm-publish instead? Less stuff for us to maintain?

Reply:
The idea would be to not use external actions to be able to check a security option to github, thus to have total control over the release pipeline

Eventually we could fork the action

Reply:
Fair point. I would prefer to fork a community one rather than maintaining ones we've written from scratch. It has a lower cost of ownership - we can merge in upstream bugfixes as needed.

Reply:
Agree
But we should be able to understand the code and what it does

(yes, I may be a little bit paranoid 😅)

Reply:
Hacking a widely used release plugin could provide attackers with an attack-surface where they can distribute malicious code to a lot of people. It's good to be paranoid about this.

These actions are fairly small and I think the effort required to understand them is smaller than the effort required to write them.

Reply:
I think that this 3rd party one is way over complex for what we need. It has billions of bells and whistles, and lots of reasons to change. i like ours better.